.github/workflows/docker.yml

```diff
@@ -168,67 +110,159 @@ jobs:
+
+      - name: Override the buildConfig variables
+        run: |
+          jq --arg PLATFORMS "$DOCKER_PLATFORMS" \
+            '. * {"agoricSyntheticChain": {
+              "platforms": ($PLATFORMS | split(","))
+            }}' package.json > ./package.json.new
+          mv ./package.json.new ./package.json
+
+          jq --arg SDK_TAG "$SDK_TAG" \
+            '. * {"agoricProposal": {
+              "sdkImageTag": $SDK_TAG,
+            }}' proposals/n\:upgrade-next/package.json > ./package.json.new
+          mv ./package.json.new proposals/n\:upgrade-next/package.json
+        working-directory: a3p-integration
+
       - name: Prefix tags
         id: prefix-tags
         run: |
-          IMAGE="$REGISTRY/agoric/agoric-sdk"
-          for TAG in ibc-alpha; do
-            PREFIXED="$PREFIXED$sep$IMAGE:$TAG"
-            sep=,
+          # These tags don't have an image name attached yet and are JSONified
+          # with newline separators because they are used as input to
+          # docker/metadata-action.
+          PREFIX=a3p-use-upgrade-next-
+          JSON_TAGS=
+          sep=
+          for TAG in "$SDK_TAG" ${{ needs.docker-sdk.outputs.tags }}; do
+            JSON_TAGS="$JSON_TAGS$sep$PREFIX$TAG"
+            sep='\n'
           done
-          echo "tags=$PREFIXED" >> $GITHUB_OUTPUT
-      - name: Set up QEMU
-        uses: docker/setup-qemu-action@v2
-      - name: Set up Docker Buildx
-        uses: docker/setup-buildx-action@v2
+          echo "json-tags=\"$JSON_TAGS\"" >> $GITHUB_OUTPUT
+
+      - name: Prepare Docker config
+        run: |
+          # prepare files for bake-action
+          yarn build:submissions
+          yarn synthetic-chain prepare-build
+
+          # Force use- targets to have our default tags by commenting out the
+          # existing definition.
+          sed -i.orig -e 's%\( tags = .*use-\)% //\1%' docker-bake.hcl
+        working-directory: a3p-integration
+
+      - uses: depot/setup-action@v1
         with:
-          buildkitd-flags: --debug
+          oidc: true # to set DEPOT_TOKEN for later steps
+      - run: depot configure-docker
       - name: Login to Docker Registry
-        uses: docker/login-action@v2
+        uses: docker/login-action@v3
         with:
           username: ${{ github.actor }}
           password: ${{ secrets.GITHUB_TOKEN }}
           registry: ${{ env.REGISTRY }}
-      - name: Build and Push
-        uses: docker/build-push-action@v4
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@v5
         with:
-          file: packages/deployment/Dockerfile.ibc-alpha
-          context: packages/deployment/docker
-          platforms: linux/amd64,linux/arm64/v8
+          images: ${{ env.REGISTRY }}/agoric/agoric-sdk
+          tags: ${{ fromJSON(steps.prefix-tags.outputs.json-tags) }}
+
+      - name: Publish "use-upgrade-next" image
```

Comment:
mind separating "ci(docker): use Depot multiarch, publish a3p-use-upgrade-next" into separate commits?

even better, separate PRs. One that closes https://github.com/Agoric/agoric-sdk/issues/9043 and another that does this publishing. There's more design to work through on that. I'd really like to see an issue which details the problem to be solved.